internal/pkg/template/diff/diff_test.go

```diff
@@ -127,6 +127,96 @@ func TestFrom_Parse(t *testing.T) {
 			},
 		},
 		"change a list item value": {},
+		"change a map to scalar": {
+			curr: `Mary:
+  Dialogue: "Said bear: 'I know I'm supposed to keep an eye on you"`,
+			old: `Mary:
+  Dialogue:
+    Bear: "I know I'm supposed to keep an eye on you"`,
+			wanted: func() *Node {
+				/* sentinel -> Mary -> Dialogue --> {new: map, old: scalar} */
+				leafDialogue := &Node{
+					key:      "Dialogue",
+					newValue: yamlScalarNode("Said bear: 'I know I'm supposed to keep an eye on you", withStyle(yaml.DoubleQuotedStyle)),
+					oldValue: yamlNode("Bear: \"I know I'm supposed to keep an eye on you\"", t),
+				}
+				return &Node{
+					children: map[string]*Node{
+						"Mary": {
+							key: "Mary",
+							children: map[string]*Node{
+								"Dialogue": leafDialogue,
+							},
+						},
+					},
+				}
+			},
+		},
+		"change a list to scalar": {
+			curr: `Mary:
+  Dialogue: "Said bear: 'I know I'm supposed to keep an eye on you; Said Dog: 'ikr'"`,
+			old: `Mary:
+  Dialogue:
+    - Bear: "I know I'm supposed to keep an eye on you"
+      Tone: disappointed
+    - Dog: "ikr"
+      Tone: pleased`,
+			wanted: func() *Node {
+				/* sentinel -> Mary -> Dialogue --> {new: list, old: scalar} */
+				leafDialogue := &Node{
+					key:      "Dialogue",
+					newValue: yamlScalarNode("Said bear: 'I know I'm supposed to keep an eye on you; Said Dog: 'ikr'", withStyle(yaml.DoubleQuotedStyle)),
+					oldValue: yamlNode(`- Bear: "I know I'm supposed to keep an eye on you"
+  Tone: disappointed
+- Dog: "ikr"
+  Tone: pleased`, t),
+				}
+				return &Node{
+					children: map[string]*Node{
+						"Mary": {
+							key: "Mary",
+							children: map[string]*Node{
+								"Dialogue": leafDialogue,
+							},
+						},
+					},
+				}
+			},
+		},
+		"change a map to list": {
+			curr: `Mary:
+  Dialogue:
+    - Bear: "I know I'm supposed to keep an eye on you"
+      Tone: disappointed
+    - Dog: "ikr"
+      Tone: pleased`,
+			old: `Mary:
+  Dialogue:
+    Bear: (disappointed) "I know I'm supposed to keep an eye on you"
+    Dog: (pleased) "ikr"`,
+			wanted: func() *Node {
```

Comment:
Why do we use `func()`?

Reply:
Without `func()` it'd have to be super deeply nested, whereas now I can define `leaf` var and reduce the nesting. It's just a matter of readability.